app/controllers/warehouse_reports/health/housing_status_controller.rb

```diff
@@ -0,0 +1,87 @@
+###
+# Copyright 2016 - 2019 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/master/LICENSE.md
+###
+
+module WarehouseReports::Health
+  class HousingStatusController < ApplicationController
+    before_action :require_can_administer_health!
+
+    def index
+      @end_date = params.dig(:filter, :end_date) || Date.today
+      @start_date = params.dig(:filter, :start_date) || @end_date - 1.month
+      @aco = params.dig(:filter, :aco)&.select{|id| id.present? }
+
+      if @start_date.to_date > @end_date.to_date
+        new_start = @end_date
+        @end_date = @start_date
+        @start_date = new_start
+      end
+
+      @report = report_data
+    end
+
+    def report_data
+      patients_scope = Health::Patient.
+        participating.
+        joins(:patient_referral)
+
+      patients_scope = patients_scope.where(patient_referrals: {accountable_care_organization_id: @aco}) if @aco.present?
+
+      patients = patients_scope.
+        preload(:sdh_case_management_notes, :epic_case_notes).
+        select(:id, :client_id, :housing_status_timestamp, :housing_status, :medicaid_id)
+
+      from_patients = patients.select { |patient| patient.housing_status_timestamp.present? }.
```

Comment:
Wouldn't this be way more efficient as `patients.where.not(housing_status_timestamp: nil)`?  This brings down all patients for the ACO and all their associated bits and then throws out a bunch using ruby.